common/caller.go

```diff
@@ -0,0 +1,29 @@
+package common
+
+import (
+	"runtime"
+	"strings"
+)
+
+func callerInfo(i int) (string, string, int, bool) {
+	ptr, file, line, ok := runtime.Caller(i)
+	var fName string
+	if ok {
+		f := runtime.FuncForPC(ptr)
+		if f != nil {
+			// f.Name() is like: github.com/docker/libnetwork/common.MethodName
+			tmp := strings.Split(f.Name(), ".")
+			if len(tmp) > 0 {
+				fName = tmp[len(tmp)-1]
+			}
+		}
+	}
+
+	return fName, file, line, ok
+}
+
+// CallerName returns the name of the function at the specified level with additional information
+// level == 0 means current method name
+func CallerName(level int) (string, string, int, bool) {
```

Comment:
CallerName is not coming from the runtime, but is simply using some of the runtime functions, so I should be verified that the method results respects the method description